src/util/include/util/move_only_function.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Xournal++
+ *
+ * Xournal util functions
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
```

Comment:
My 2 cents:

Is this implementation a ported version of the C++23 [move_only_function](https://en.cppreference.com/w/cpp/utility/functional/move_only_function) ?
If that is the case, are any differences between this and C++23 impl?

I would like to see a documentation of what it does and how to use it plus a link to C++23 [move_only_function](https://en.cppreference.com/w/cpp/utility/functional/move_only_function) (if applies).
I think it will increase the chances that a future developer use this kind of helpers.

Reply:
Thanks!